### ISSUE TITLE
add timeout before create the warning

### ISSUE THREAD

**Issue by azlux** [firetech/wifiwarning#4]
Without timeout, when we start our wifi, the notification show up before the wifi connect.

**Comment by firetech** [comment]
That is possible, but would (unless I'm mistaken) add a relatively large amount of complexity to the backend.

That the notification shows up after enabling wifi is intended behavior. I myself interpret it as a signal that wifi was enabled properly. The intention is that, at all time when wifi is active,  either the notification or the wifi signal status is shown in the status bar. This is mostly true.

**Comment by firetech** [comment]
Closing this for now. I might reopen if someone makes a compelling argument.

**Comment by azlux** [comment]
I'm not an expert on Android. Not sure if it's help
https://developer.android.com/reference/android/os/Handler.html#postDelayed(java.lang.Runnable,%20long)

**Comment by firetech** [comment]
Hmm... That could work, yes. I have next to no free time, though, so it won't be implemented soon...

**Comment by robinpaulson** [comment]
Yeah, I think this is pretty important. I was surprised the first time I used this that it beeped at me when I turned on Wifi in an area with a known hotspot. And it does that every time I turn it on! Very irritating; I would expect it to wait 10 seconds or so and *then* alert me if there is no connection.

I'm reminded of a principal in security that says if you constantly ask users for permission to do something, eventually they stop reading the messages and click "Yes" for everything, or turn off the security feature. Similarly, I stopped using this software after the 10th or so beep.

I'm sure the code would be trivial, merely putting a sleep(10) (or the java equivalent) in an appropriate place would do it.

**Comment by robinpaulson** [comment]
@firetech Is there any intention to add this suggestion, perhaps as a selectable option? Would that decision rely upon providing a pull request?

**Comment by firetech** [comment]
@robinpaulson If someone makes a working pull request, I might add it. I've had no time to do this myself so far.

I understand that it may be slightly annoying when having a ringtone or vibration set on the notification. Those settings were added on request and weren't really my intention from the beginning. I use the app with a completely silent notification myself, just to see if WiFi is enabled but not connected when looking at my phone (kind of like the status icons next to the clock on stock Android).
